python/py-gwpy/Portfile

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+PortGroup           github 1.0
+
+github.setup        gwpy gwpy 0.7.0 v
+name                py-gwpy
+
+categories-append   science
+maintainers         {ligo.org:duncan.macleod @duncanmmacleod}
+
+platforms           darwin
+license             GPL-3
+
+description         A python package for gravitational-wave astrophysics
+long_description    GWpy is a collaboration-driven Python package providing \
+                    tools for studying data from ground-based \
+                    gravitational-wave detectors.
+homepage            https://gwpy.github.io
+
+master_sites        pypi:g/gwpy \
+                    https://github.com/gwpy/gwpy/releases/v${version}/
+distname            gwpy-${version}
+
+checksums   rmd160  6068b4ad15f3b3a4b615605c3d22196fb042396f \
+            sha256  6ced45ec9d62f60e01863a5ffb481d01b0f8af40333b99c0efd6db9c0f9a80f0
+
+# until glue is available for multiple python versions, we are stuck to
+# python 2.7
+#python.versions     27 34 35 36
+python.versions     27
+python.default_version 27
+
+if {${name} ne ${subport}} {
+
+    depends_lib-append  port:py${python.version}-six \
+                        port:py${python.version}-dateutil \
+                        port:py${python.version}-numpy \
+                        port:py${python.version}-scipy \
+                        port:py${python.version}-matplotlib \
+                        port:py${python.version}-astropy \
+                        port:py${python.version}-lal \
+                        port:glue
+    if {${python.version} < 34} {
+        depends_lib-append port:py${python.version}-enum34
+    }
+
+    depends_build-append port:py${python.version}-setuptools
```

Comment:
For all of the variants, you only add a dependency. What if the user has the port(s) installed but does not have the variant selected? The variant should disable the support if it is not selected.

Reply:
I'm not sure how to effectively address this comment. The code is just python where certain extra class methods will work with optional dependencies, or raise `ImportError` when those are missing.